test-lit/lit_port.py

```diff
@@ -0,0 +1,256 @@
+""" lit_port.py
+
+    Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+    See https://llvm.org/LICENSE.txt for license information.
+    SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+"""
+
+import argparse
+import os
+import re
+import shlex
+import sys
+import subprocess
+from pathlib import Path
+
+from test_splitter import TestSplitter
+
+ROOT = Path.cwd()
+while ROOT.name != "f18":
+    ROOT = ROOT.parent
+DEFAULT_TEMPLATE = "! RUN: %flang {} %s 2>&1\n"
```

Comment:
Is the redirection "2>&1" necessary here?

Reply:
Trying to emulate the shell scripts used (i.e test_errors.sh) suggests yes, but I'm not 100% sure. I suspect yes, since error tests will need piping from stderr under normal circumstances, but I'm unsure if lit is able to automatically capture all output without this pipe.